### DISCUSSION TITLE
Load & Install Extensions for JDBC connections

### DISCUSSION THREAD

**Discussion by hu6360567** [duckdb/duckdb#6901] [Ideas]
Why do you want this feature?
I'm trying to implement some UDF with DuckDB, while the application is implemented in Java. As [#2670 ](https://github.com/duckdb/duckdb/issues/2670#issuecomment-978852102) refers, we can add implement UDFs into C++ codebase and recompile the JDBC driver.
It seems unacceptable for my current development.

Is it possible to implement UDFs in C++ and compile it into an extension to install and load when JDBC connection is created, so that I can maintain a much smaller codebase and less compiling time? 

For current JDBC implementation, we can set Option to C++ `DBConfig` throw Properties when getConnection
```java
  Properties prop = new Properties();
  prop.setProperty("allow_unsigned_extensions", "true");
  try (var conn = DriverManager.getConnection("jdbc:duckdb:", prop)) {
    // Blah, blah, blah ...
  }
```

~~But it seems no way to load for DBConfig when the connection is created.~~

Extensions can be loaded and installed by execute statemens
```java
try (var conn = DriverManager.getConnection("jdbc:duckdb:", prop)) {

            try (PreparedStatement stmt = conn.prepareStatement("select * from duckdb_extensions() where extension_name = 'json'");
                 DuckDBResultSet resultSet = (DuckDBResultSet) stmt.executeQuery();
                 var allocator = new RootAllocator()
            ){
                try (var reader = (ArrowReader) resultSet.arrowExportStream(allocator, 256)) {
                    while (reader.loadNextBatch()) {
                        System.out.println(reader.getVectorSchemaRoot().contentToTSVString());
                    }
                }
            }

            try(PreparedStatement stmt = conn.prepareStatement("load json")){
                stmt.execute();
            }

            try (PreparedStatement stmt = conn.prepareStatement("select * from duckdb_extensions() where extension_name = 'json'");
                 DuckDBResultSet resultSet = (DuckDBResultSet) stmt.executeQuery();
                 var allocator = new RootAllocator()
            ){
                try (var reader = (ArrowReader) resultSet.arrowExportStream(allocator, 256)) {
                    while (reader.loadNextBatch()) {
                        System.out.println(reader.getVectorSchemaRoot().contentToTSVString());
                    }
                }
            }

        }
```

**Comment by hannes** [comment]
Sorry, what's the question here?

**Reply by hu6360567** [reply]
I orginally thought extensions need to be manipulate by native codes, but it turns out that can be done by statements.

**Comment by Tishj** [comment]
Yes you can add functions as part of an extension, and you can should be able to run them with the JDBC client, as you demonstrated with the JSON extension.
